### ISSUE TITLE
Unable to exit select Object in events until clicked outside the menu

### ISSUE THREAD

**Issue by geetesh-gupta** [4ian/GDevelop#1558]
## Describe the bug
Whenever we try to select an object for an event in the Events tab, we cannot exit the select object menu until clicked outside the menu. The expected behaviour should be whenever an object is selected, 

- if "Enter" is pressed, the object should be selected and the menu should disappear.
- if "Esc" is pressed, the previous value of the object is selected and the menu should disappear.

Also, this will make it consistent with other triggers like Comment and Group.

## To Reproduce
Steps to reproduce the behaviour:
1. Open any project
2. Click on the Events tab
3. Scroll down to any event wherever an object is desired
4. Click on to change the object
5. Select any object
6. See error (unable to exit until clicked somewhere else)
![image](https://user-images.githubusercontent.com/34645022/77136974-dc79e680-6a91-11ea-827f-8b9e920668b9.png)

**Comment by 4ian** [comment]
We talked about this with @HarsimranVirk in the implementation of the new AutoComplete.
Would be interesting to add this, surely by plugging together a bunch of "onRequestClose" and add a few things in SemiControlledAutocomplete.

> if "Enter" is pressed, the object should be selected and the menu should disappear.
> if "Esc" is pressed, the previous value of the object is selected and the menu should disappear.

This sounds correct.

**Comment by HarsimranVirk** [comment]
@4ian we had this discussion that we might be able to achieve this in the inlinePopover element itself.
So, I've tried to use an onKeyDown listener on inlinePopover, and it works well only while we press enter for selecting the options (ie, not when a mouse click is used). Also, I'm afraid that it wouldn't be able to guarantee the selection of an option in all the cases. 
How would you suggest to proceed with this? Should we pass down the onRequestClose() as a prop? Also, you said something like, "events are bubbled up down the tree", I didn't really understand that part.

**Comment by 4ian** [comment]
> So, I've tried to use an onKeyDown listener on inlinePopover, and it works well only while we press enter for selecting the options (ie, not when a mouse click is used). Also, I'm afraid that it wouldn't be able to guarantee the selection of an option in all the cases.

As you said, there is a high risk that this won't guarantee that the selection is properly done.
This is because the InlinePopover would be conflicting/catching the click/key press before the component (input/text field) has a chance to process it. It's also not super deterministic and could change according to the browser or the OS.

> Should we pass down the onRequestClose() as a prop? 

Yes :) This is equivalent to say to the child component: "here is a prop, please call it when you detect that the user wants to close the selection. I don't personally know what the user has to do, but I'm confident that you, the child component, will call this prop as the right time" :) 
Which is ok I think.

Basically, we're making the fields responsible for calling onRequestClose when they know that the user made a selection.
Note that the fields can be used WITHOUT passing onRequestClose. If there is nothing to close (like when you're editing the action/condition entirely, in NewInstructionEditorDialog).

> Also, you said something like, "events are bubbled up down the tree", I didn't really understand that part.

In the DOM rendered by the browser, an event is first sent to the most deep div/element, then if not stopped is then triggered on the parent, etc..
